### ISSUE TITLE
[psqldef] parse error when cast function call result to other type

### ISSUE THREAD

**Issue by sasurau4** [k0kubun/sqldef#233]
## Overview

I met parse error when trying to cast function call result

## Input SQL

```sql
CREATE TABLE public.test_table (
    "id" uuid NOT NULL,
    "tenant_id" uuid NOT NULL,
    PRIMARY KEY ("id", "tenant_id")
);
CREATE POLICY tenant_isolation_policy ON test_table AS PERMISSIVE FOR ALL TO public USING current_setting('app.current_tenant')::uuid);
```

Same error when using `CAST` and `CONVERT` instead of short syntax `::uuid`

## Current output

```
found syntax error when parsing DDL "CREATE POLICY tenant_isolation_policy ON test_table AS PERMISSIVE FOR ALL TO public USING current_setting('app.current_tenant')::uuid)": syntax error at position 134 near 'uuid'
```

## Expected output

No error with the input

**Comment by k0kubun** [comment]
I managed to write an implementation that parses this correctly, but is that a valid input? I also get an error on PostgreSQL 14.1:

```sql
test=# CREATE TABLE public.test_table (
    "id" uuid NOT NULL,
    "tenant_id" uuid NOT NULL,
    PRIMARY KEY ("id", "tenant_id")
);
CREATE TABLE
test=# CREATE POLICY tenant_isolation_policy ON test_table AS PERMISSIVE FOR ALL TO public USING current_setting('app.current_tenant')::uuid);
ERROR:  syntax error at or near "current_setting"
LINE 1: ... test_table AS PERMISSIVE FOR ALL TO public USING current_se...
                                                             ^
```

**Comment by k0kubun** [comment]
Ah, I guess I should call `set_config`. nvm

**Comment by k0kubun** [comment]
https://www.postgresql.org/docs/current/sql-createpolicy.html
`using_expression` seems to take a boolean expression, so `USING current_setting('app.current_tenant')::uuid)` is invalid, even if I ignore the last `)`.

**Comment by k0kubun** [comment]
I can't verify your input since it's just invalid, but I believe the issue was fixed in v0.11.54.
